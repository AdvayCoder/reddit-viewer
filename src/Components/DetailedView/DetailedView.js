import { Flex, Text, Box, Image, Skeleton, Center } from '@chakra-ui/react';
import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { loadSinglePost, selectSinglePost, selectIsLoading, selectSinglePostError } from "../../features/posts/postsSlice";
import { useSelector, useDispatch } from "react-redux";
import checkFileType from '../../utils/checkFileType';
import Comment from '../Comment/Comment';
import PageNotFound from "../../Components/PageNotFound/PageNotFound";

const DetailedView = () => {

    const { id } = useParams();

    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(loadSinglePost(`https://www.reddit.com/r/ProgrammerHumor/comments/${id}.json`));
    }, [dispatch]);

    const singlePostData = useSelector(selectSinglePost);
    const isLoading = useSelector(selectIsLoading);
    const singlePostError = useSelector(selectSinglePostError);

    if(singlePostError) {
        return <PageNotFound />;
    }

    const PostSeleton = () => {
        return (
            <Flex mt='100px' w='100%' flexDir='column' align='center'>
                <Flex flexDir='column' align='center' sx={{flexGrow: 1}}>
                    <Skeleton my='0.5rem' h='10px' w='80px' />
                    <Skeleton my='0.5rem' h='45px' w='230px' />
                    <Skeleton my='0.5rem' h='400px' w='400px' />
                </Flex>
                <Flex ml='2em' mt='1em' flexDir='column' my='5rem' sx={{flexGrow: 1}}>
                        <Skeleton h='50rem' w='80vw' />
                </Flex>
            </Flex>
        )
    }

    return (
        <Flex>
            {isLoading ? (
                <PostSeleton />
            ) : (
                Object.values(singlePostData).length && (
                    <Flex mt='100px' maxW='100%' w='100%' flexDir='column'>
                        <Flex flexDir='column' align='center' sx={{flexGrow: 1}}>
                            <Text color='#2D2F39' fontWeight={600}>u/{singlePostData.main.author}</Text>
                            <Text fontWeight={700} fontSize={['1.5rem', null, '2rem', null]} maxW='700px'>{singlePostData.main.title}</Text>
                            {checkFileType(singlePostData.main.resourceUrl) === 'image' ? (
                                <Image src={singlePostData.main.resourceUrl} maxW='60%' maxH='80%' />
                            ) : (
                                checkFileType(singlePostData.main.resourceUrl) === 'video' ? (
                                    <Box as='video' src={singlePostData.main.resourceUrl} maxW='60%' maxH='80%' />
                                ) : (
                                    <Text>Error loading the file!</Text>
                                )
                            )}
                        </Flex>
                        <Flex ml='2em' mt='1em' flexDir='column' sx={{flexGrow: 1}}>
                                <Text fontWeight={600} align='center'>Comments ({singlePostData.comments.length})</Text>
                                {singlePostData.comments.map((comment, index) => <Comment author={comment.author} body={comment.body} key={index} />)}
                        </Flex>
                    </Flex>
                )  
            )}
        </Flex>
    )
}

export default DetailedView;