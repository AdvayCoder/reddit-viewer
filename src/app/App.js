import React, { useEffect } from 'react';
import './App.css';
import { BrowserRouter as Router, Routes, Route} from 'react-router-dom';
import NavBar from '../Components/NavBar/NavBar';
import HomePage from '../pages/HomePage/HomePage';
import InitFonts from './InitFonts/InitFonts';
import { useDispatch } from 'react-redux';
import { loadPosts } from '../features/posts/postsSlice';
import DetailedView from '../Components/DetailedView/DetailedView';

function App() {

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadPosts('https://www.reddit.com/r/ProgrammerHumor.json?limit=100'));
  }, [dispatch]);

  return (
    <Router>
      <InitFonts />
      <NavBar />
      <Routes>
        <Route path="/" element={ <HomePage /> } />
        <Route path={'/post/:id'} element={<DetailedView />} />
      </Routes>
    </Router>
  );
}

export default App;
