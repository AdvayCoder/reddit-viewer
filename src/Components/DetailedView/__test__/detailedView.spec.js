import DetailedView from "../DetailedView";
import { Provider } from "react-redux";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { store } from '../../../app/store'
import { mount, shallow, configure } from "enzyme";

configure({ adapter: new Adapter})

describe('DetailedView', () => {
    it('renders properly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <DetailedView />
            </Provider>
        )

        expect(wrapper).toMatchSnapshot();
    })

    // it('shows data correctly', () => {
    //     const wrapper = mount(
    //         <Provider store={store}>
    //             <DetailedView />
    //         </Provider>
    //     )

        
    // })
})