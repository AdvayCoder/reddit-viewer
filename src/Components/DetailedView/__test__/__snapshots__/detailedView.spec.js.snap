// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedView renders properly 1`] = `ShallowWrapper {}`;
